Tests errors on Python 3.8 (GIT master)

Python 3.8beta3 with commit fd585ce36faba338fa1c71949cf29ed8e8f5ea19

Environment (cloupickle is unbundled):
```
DEBUG util.py:587:   python3-cloudpickle         noarch       1.2.1-3.fc32
DEBUG util.py:587:   python3-devel               armv7hl      3.8.0~b3-4.fc32
DEBUG util.py:587:   python3-lz4                 armv7hl      2.1.2-5.fc32
DEBUG util.py:587:   python3-numpy               armv7hl      1:1.17.0-3.fc32
DEBUG util.py:587:   python3-psutil              armv7hl      5.6.3-3.fc32
DEBUG util.py:587:   python3-pytest              noarch       4.6.5-3.fc32 
```

There seems to be a hash issue:

```
=================================== FAILURES ===================================
________ test_hashes_stay_the_same[This is a string to hash-expected0] _________
to_hash = 'This is a string to hash'
expected = {'py2': '80436ada343b0d79a99bfd8883a96e45', 'py3': '71b3f47df22cb19431d85d92d0b230b2'}
    @parametrize('to_hash,expected',
                 [('This is a string to hash',
                     {'py2': '80436ada343b0d79a99bfd8883a96e45',
                      'py3': '71b3f47df22cb19431d85d92d0b230b2'}),
                  (u"C'est l\xe9t\xe9",
                     {'py2': '2ff3a25200eb6219f468de2640913c2d',
                      'py3': '2d8d189e9b2b0b2e384d93c868c0e576'}),
                  ((123456, 54321, -98765),
                     {'py2': '50d81c80af05061ac4dcdc2d5edee6d6',
                      'py3': 'e205227dd82250871fa25aa0ec690aa3'}),
                  ([random.Random(42).random() for _ in range(5)],
                     {'py2': '1a36a691b2e2ba3a9df72de3dccf17ea',
                      'py3': 'a11ffad81f9682a7d901e6edc3d16c84'}),
                  ([3, 'abc', None, TransportableException('foo', ValueError)],
                     {'py2': 'adb6ba84990ee5e462dc138383f11802',
                      'py3': '994f663c64ba5e64b2a85ebe75287829'}),
                  ({'abcde': 123, 'sadfas': [-9999, 2, 3]},
                     {'py2': 'fc9314a39ff75b829498380850447047',
                      'py3': 'aeda150553d4bb5c69f0e69d51b0e2ef'})])
    def test_hashes_stay_the_same(to_hash, expected):
        # We want to make sure that hashes don't change with joblib
        # version. For end users, that would mean that they have to
        # regenerate their cache from scratch, which potentially means
        # lengthy recomputations.
        # Expected results have been generated with joblib 0.9.2
    
        py_version_str = 'py3' if PY3_OR_LATER else 'py2'
>       assert hash(to_hash) == expected[py_version_str]
E       AssertionError: assert '115916c98f74...58a5c7184c94c' == '71b3f47df22cb...85d92d0b230b2'
E         - 115916c98f747117d0458a5c7184c94c
E         + 71b3f47df22cb19431d85d92d0b230b2
joblib/test/test_hashing.py:372: AssertionError
____________ test_hashes_stay_the_same[C'est l\xe9t\xe9-expected1] _____________
to_hash = "C'est lété"
expected = {'py2': '2ff3a25200eb6219f468de2640913c2d', 'py3': '2d8d189e9b2b0b2e384d93c868c0e576'}
    @parametrize('to_hash,expected',
                 [('This is a string to hash',
                     {'py2': '80436ada343b0d79a99bfd8883a96e45',
                      'py3': '71b3f47df22cb19431d85d92d0b230b2'}),
                  (u"C'est l\xe9t\xe9",
                     {'py2': '2ff3a25200eb6219f468de2640913c2d',
                      'py3': '2d8d189e9b2b0b2e384d93c868c0e576'}),
                  ((123456, 54321, -98765),
                     {'py2': '50d81c80af05061ac4dcdc2d5edee6d6',
                      'py3': 'e205227dd82250871fa25aa0ec690aa3'}),
                  ([random.Random(42).random() for _ in range(5)],
                     {'py2': '1a36a691b2e2ba3a9df72de3dccf17ea',
                      'py3': 'a11ffad81f9682a7d901e6edc3d16c84'}),
                  ([3, 'abc', None, TransportableException('foo', ValueError)],
                     {'py2': 'adb6ba84990ee5e462dc138383f11802',
                      'py3': '994f663c64ba5e64b2a85ebe75287829'}),
                  ({'abcde': 123, 'sadfas': [-9999, 2, 3]},
                     {'py2': 'fc9314a39ff75b829498380850447047',
                      'py3': 'aeda150553d4bb5c69f0e69d51b0e2ef'})])
    def test_hashes_stay_the_same(to_hash, expected):
        # We want to make sure that hashes don't change with joblib
        # version. For end users, that would mean that they have to
        # regenerate their cache from scratch, which potentially means
        # lengthy recomputations.
        # Expected results have been generated with joblib 0.9.2
    
        py_version_str = 'py3' if PY3_OR_LATER else 'py2'
>       assert hash(to_hash) == expected[py_version_str]
E       AssertionError: assert '6cfa6be6cf88...faa7bbaf4e5ee' == '2d8d189e9b2b0...d93c868c0e576'
E         - 6cfa6be6cf88fc4835cfaa7bbaf4e5ee
E         + 2d8d189e9b2b0b2e384d93c868c0e576
joblib/test/test_hashing.py:372: AssertionError
________________ test_hashes_stay_the_same[to_hash2-expected2] _________________
to_hash = (123456, 54321, -98765)
expected = {'py2': '50d81c80af05061ac4dcdc2d5edee6d6', 'py3': 'e205227dd82250871fa25aa0ec690aa3'}
    @parametrize('to_hash,expected',
                 [('This is a string to hash',
                     {'py2': '80436ada343b0d79a99bfd8883a96e45',
                      'py3': '71b3f47df22cb19431d85d92d0b230b2'}),
                  (u"C'est l\xe9t\xe9",
                     {'py2': '2ff3a25200eb6219f468de2640913c2d',
                      'py3': '2d8d189e9b2b0b2e384d93c868c0e576'}),
                  ((123456, 54321, -98765),
                     {'py2': '50d81c80af05061ac4dcdc2d5edee6d6',
                      'py3': 'e205227dd82250871fa25aa0ec690aa3'}),
                  ([random.Random(42).random() for _ in range(5)],
                     {'py2': '1a36a691b2e2ba3a9df72de3dccf17ea',
                      'py3': 'a11ffad81f9682a7d901e6edc3d16c84'}),
                  ([3, 'abc', None, TransportableException('foo', ValueError)],
                     {'py2': 'adb6ba84990ee5e462dc138383f11802',
                      'py3': '994f663c64ba5e64b2a85ebe75287829'}),
                  ({'abcde': 123, 'sadfas': [-9999, 2, 3]},
                     {'py2': 'fc9314a39ff75b829498380850447047',
                      'py3': 'aeda150553d4bb5c69f0e69d51b0e2ef'})])
    def test_hashes_stay_the_same(to_hash, expected):
        # We want to make sure that hashes don't change with joblib
        # version. For end users, that would mean that they have to
        # regenerate their cache from scratch, which potentially means
        # lengthy recomputations.
        # Expected results have been generated with joblib 0.9.2
    
        py_version_str = 'py3' if PY3_OR_LATER else 'py2'
>       assert hash(to_hash) == expected[py_version_str]
E       AssertionError: assert '6f7adce43153...a73430e3c79a3' == 'e205227dd8225...25aa0ec690aa3'
E         - 6f7adce43153280d4c1a73430e3c79a3
E         + e205227dd82250871fa25aa0ec690aa3
joblib/test/test_hashing.py:372: AssertionError
________________ test_hashes_stay_the_same[to_hash3-expected3] _________________
to_hash = [0.6394267984578837, 0.6394267984578837, 0.6394267984578837, 0.6394267984578837, 0.6394267984578837]
expected = {'py2': '1a36a691b2e2ba3a9df72de3dccf17ea', 'py3': 'a11ffad81f9682a7d901e6edc3d16c84'}
    @parametrize('to_hash,expected',
                 [('This is a string to hash',
                     {'py2': '80436ada343b0d79a99bfd8883a96e45',
                      'py3': '71b3f47df22cb19431d85d92d0b230b2'}),
                  (u"C'est l\xe9t\xe9",
                     {'py2': '2ff3a25200eb6219f468de2640913c2d',
                      'py3': '2d8d189e9b2b0b2e384d93c868c0e576'}),
                  ((123456, 54321, -98765),
                     {'py2': '50d81c80af05061ac4dcdc2d5edee6d6',
                      'py3': 'e205227dd82250871fa25aa0ec690aa3'}),
                  ([random.Random(42).random() for _ in range(5)],
                     {'py2': '1a36a691b2e2ba3a9df72de3dccf17ea',
                      'py3': 'a11ffad81f9682a7d901e6edc3d16c84'}),
                  ([3, 'abc', None, TransportableException('foo', ValueError)],
                     {'py2': 'adb6ba84990ee5e462dc138383f11802',
                      'py3': '994f663c64ba5e64b2a85ebe75287829'}),
                  ({'abcde': 123, 'sadfas': [-9999, 2, 3]},
                     {'py2': 'fc9314a39ff75b829498380850447047',
                      'py3': 'aeda150553d4bb5c69f0e69d51b0e2ef'})])
    def test_hashes_stay_the_same(to_hash, expected):
        # We want to make sure that hashes don't change with joblib
        # version. For end users, that would mean that they have to
        # regenerate their cache from scratch, which potentially means
        # lengthy recomputations.
        # Expected results have been generated with joblib 0.9.2
    
        py_version_str = 'py3' if PY3_OR_LATER else 'py2'
>       assert hash(to_hash) == expected[py_version_str]
E       AssertionError: assert '889a464e6239...e0306ad9b15e5' == 'a11ffad81f968...1e6edc3d16c84'
E         - 889a464e6239c2e7a64e0306ad9b15e5
E         + a11ffad81f9682a7d901e6edc3d16c84
joblib/test/test_hashing.py:372: AssertionError
________________ test_hashes_stay_the_same[to_hash4-expected4] _________________
to_hash = [3, 'abc', None, TransportableException
___________________________________________________________________________
foo
___________________________________________________________________________]
expected = {'py2': 'adb6ba84990ee5e462dc138383f11802', 'py3': '994f663c64ba5e64b2a85ebe75287829'}
    @parametrize('to_hash,expected',
                 [('This is a string to hash',
                     {'py2': '80436ada343b0d79a99bfd8883a96e45',
                      'py3': '71b3f47df22cb19431d85d92d0b230b2'}),
                  (u"C'est l\xe9t\xe9",
                     {'py2': '2ff3a25200eb6219f468de2640913c2d',
                      'py3': '2d8d189e9b2b0b2e384d93c868c0e576'}),
                  ((123456, 54321, -98765),
                     {'py2': '50d81c80af05061ac4dcdc2d5edee6d6',
                      'py3': 'e205227dd82250871fa25aa0ec690aa3'}),
                  ([random.Random(42).random() for _ in range(5)],
                     {'py2': '1a36a691b2e2ba3a9df72de3dccf17ea',
                      'py3': 'a11ffad81f9682a7d901e6edc3d16c84'}),
                  ([3, 'abc', None, TransportableException('foo', ValueError)],
                     {'py2': 'adb6ba84990ee5e462dc138383f11802',
                      'py3': '994f663c64ba5e64b2a85ebe75287829'}),
                  ({'abcde': 123, 'sadfas': [-9999, 2, 3]},
                     {'py2': 'fc9314a39ff75b829498380850447047',
                      'py3': 'aeda150553d4bb5c69f0e69d51b0e2ef'})])
    def test_hashes_stay_the_same(to_hash, expected):
        # We want to make sure that hashes don't change with joblib
        # version. For end users, that would mean that they have to
        # regenerate their cache from scratch, which potentially means
        # lengthy recomputations.
        # Expected results have been generated with joblib 0.9.2
    
        py_version_str = 'py3' if PY3_OR_LATER else 'py2'
>       assert hash(to_hash) == expected[py_version_str]
E       AssertionError: assert '71fef3696bd5...e7bbf050dbfda' == '994f663c64ba5...85ebe75287829'
E         - 71fef3696bd507b1e36e7bbf050dbfda
E         + 994f663c64ba5e64b2a85ebe75287829
joblib/test/test_hashing.py:372: AssertionError
________________ test_hashes_stay_the_same[to_hash5-expected5] _________________
to_hash = {'abcde': 123, 'sadfas': [-9999, 2, 3]}
expected = {'py2': 'fc9314a39ff75b829498380850447047', 'py3': 'aeda150553d4bb5c69f0e69d51b0e2ef'}
    @parametrize('to_hash,expected',
                 [('This is a string to hash',
                     {'py2': '80436ada343b0d79a99bfd8883a96e45',
                      'py3': '71b3f47df22cb19431d85d92d0b230b2'}),
                  (u"C'est l\xe9t\xe9",
                     {'py2': '2ff3a25200eb6219f468de2640913c2d',
                      'py3': '2d8d189e9b2b0b2e384d93c868c0e576'}),
                  ((123456, 54321, -98765),
                     {'py2': '50d81c80af05061ac4dcdc2d5edee6d6',
                      'py3': 'e205227dd82250871fa25aa0ec690aa3'}),
                  ([random.Random(42).random() for _ in range(5)],
                     {'py2': '1a36a691b2e2ba3a9df72de3dccf17ea',
                      'py3': 'a11ffad81f9682a7d901e6edc3d16c84'}),
                  ([3, 'abc', None, TransportableException('foo', ValueError)],
                     {'py2': 'adb6ba84990ee5e462dc138383f11802',
                      'py3': '994f663c64ba5e64b2a85ebe75287829'}),
                  ({'abcde': 123, 'sadfas': [-9999, 2, 3]},
                     {'py2': 'fc9314a39ff75b829498380850447047',
                      'py3': 'aeda150553d4bb5c69f0e69d51b0e2ef'})])
    def test_hashes_stay_the_same(to_hash, expected):
        # We want to make sure that hashes don't change with joblib
        # version. For end users, that would mean that they have to
        # regenerate their cache from scratch, which potentially means
        # lengthy recomputations.
        # Expected results have been generated with joblib 0.9.2
    
        py_version_str = 'py3' if PY3_OR_LATER else 'py2'
>       assert hash(to_hash) == expected[py_version_str]
E       AssertionError: assert 'b8d5ad181711...9dc9308b8ae05' == 'aeda150553d4b...0e69d51b0e2ef'
E         - b8d5ad18171111167d69dc9308b8ae05
E         + aeda150553d4bb5c69f0e69d51b0e2ef
joblib/test/test_hashing.py:372: AssertionError
_________________ test_hashes_stay_the_same_with_numpy_objects _________________
    @with_numpy
    def test_hashes_stay_the_same_with_numpy_objects():
        # We want to make sure that hashes don't change with joblib
        # version. For end users, that would mean that they have to
        # regenerate their cache from scratch, which potentially means
        # lengthy recomputations.
        rng = np.random.RandomState(42)
        # Being explicit about dtypes in order to avoid
        # architecture-related differences. Also using 'f4' rather than
        # 'f8' for float arrays because 'f8' arrays generated by
        # rng.random.randn don't seem to be bit-identical on 32bit and
        # 64bit machines.
        to_hash_list = [
            rng.randint(-1000, high=1000, size=50).astype('<i8'),
            tuple(rng.randn(3).astype('<f4') for _ in range(5)),
            [rng.randn(3).astype('<f4') for _ in range(5)],
            {
                -3333: rng.randn(3, 5).astype('<f4'),
                0: [
                    rng.randint(10, size=20).astype('<i8'),
                    rng.randn(10).astype('<f4')
                ]
            },
            # Non regression cases for https://github.com/joblib/joblib/issues/308.
            # Generated with joblib 0.9.4.
            np.arange(100, dtype='<i8').reshape((10, 10)),
            # Fortran contiguous array
            np.asfortranarray(np.arange(100, dtype='<i8').reshape((10, 10))),
            # Non contiguous array
            np.arange(100, dtype='<i8').reshape((10, 10))[:, :2],
        ]
    
        # These expected results have been generated with joblib 0.9.0
        expected_dict = {'py2': ['80f2387e7752abbda2658aafed49e086',
                                 '0d700f7f25ea670fd305e4cd93b0e8cd',
                                 '83a2bdf843e79e4b3e26521db73088b9',
                                 '63e0efd43c0a9ad92a07e8ce04338dd3',
                                 '03fef702946b602c852b8b4e60929914',
                                 '07074691e90d7098a85956367045c81e',
                                 'd264cf79f353aa7bbfa8349e3df72d8f'],
                         'py3': ['10a6afc379ca2708acfbaef0ab676eab',
                                 '988a7114f337f381393025911ebc823b',
                                 'c6809f4b97e35f2fa0ee8d653cbd025c',
                                 'b3ad17348e32728a7eb9cda1e7ede438',
                                 '927b3e6b0b6a037e8e035bda134e0b05',
                                 '108f6ee98e7db19ea2006ffd208f4bf1',
                                 'bd48ccaaff28e16e6badee81041b7180']}
    
        py_version_str = 'py3' if PY3_OR_LATER else 'py2'
        expected_list = expected_dict[py_version_str]
    
        for to_hash, expected in zip(to_hash_list, expected_list):
>           assert hash(to_hash) == expected
E           AssertionError: assert '445fe65329ff...e94ff3d8e673b' == '10a6afc379ca2...baef0ab676eab'
E             - 445fe65329ff1c52ea9e94ff3d8e673b
E             + 10a6afc379ca2708acfbaef0ab676eab
joblib/test/test_hashing.py:447: AssertionError
_____________________ test_no_semaphore_tracker_on_import ______________________
    @pytest.mark.skipif(PY27, reason="Need python3.3+")
    def test_no_semaphore_tracker_on_import():
        # check that importing joblib does not implicitly spawn a ressource tracker
        # or a semaphore tracker
        code = """if True:
            import joblib
            from multiprocessing import semaphore_tracker
            # The following line would raise RuntimeError if the
            # start_method is already set.
            msg = "multiprocessing.semaphore_tracker has been spawned on import"
            assert semaphore_tracker._semaphore_tracker._fd is None, msg"""
>       check_subprocess_call([sys.executable, '-c', code])
joblib/test/test_module.py:38: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
cmd = ['/usr/bin/python3', '-c', 'if True:\n        import joblib\n        from multiprocessing import semaphore_tracker\n  ...g.semaphore_tracker has been spawned on import"\n        assert semaphore_tracker._semaphore_tracker._fd is None, msg']
timeout = 5, stdout_regex = None, stderr_regex = None
    def check_subprocess_call(cmd, timeout=5, stdout_regex=None,
                              stderr_regex=None):
        """Runs a command in a subprocess with timeout in seconds.
    
        Also checks returncode is zero, stdout if stdout_regex is set, and
        stderr if stderr_regex is set.
        """
        proc = subprocess.Popen(cmd, stdout=subprocess.PIPE,
                                stderr=subprocess.PIPE)
    
        def kill_process():
            warnings.warn("Timeout running {}".format(cmd))
            proc.kill()
    
        timer = threading.Timer(timeout, kill_process)
        try:
            timer.start()
            stdout, stderr = proc.communicate()
    
            if PY3_OR_LATER:
                stdout, stderr = stdout.decode(), stderr.decode()
            if proc.returncode != 0:
                message = (
                    'Non-zero return code: {}.\nStdout:\n{}\n'
                    'Stderr:\n{}').format(
                        proc.returncode, stdout, stderr)
>               raise ValueError(message)
E               ValueError: Non-zero return code: 1.
E               Stdout:
E               
E               Stderr:
E               Traceback (most recent call last):
E                 File "<string>", line 3, in <module>
E               ImportError: cannot import name 'semaphore_tracker' from 'multiprocessing' (/usr/lib/python3.8/multiprocessing/__init__.py)
joblib/testing.py:65: ValueError
=============================== warnings summary ===============================
/usr/lib/python3.8/site-packages/_pytest/mark/structures.py:330
  /usr/lib/python3.8/site-packages/_pytest/mark/structures.py:330: PytestUnknownMarkWarning: Unknown pytest.mark.timeout - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/latest/mark.html
    warnings.warn(
joblib/test/test_func_inspect.py::test_filter_args_2
  /builddir/build/BUILDROOT/python-joblib-0.13.2-5.20190824gitfd585ce.fc32.noarch/usr/lib/python3.8/site-packages/joblib/test/test_func_inspect.py:132: UserWarning: Cannot inspect object functools.partial(<function f at 0xaf812928>, 1), ignore list will not work.
    assert filter_args(ff, ['y'], (1, )) == {'*': [1], '**': {}}
joblib/test/test_func_inspect.py::test_filter_args_python_3
joblib/test/test_func_inspect.py::test_filter_args_error_msg[ValueError-Ignore list: argument \\'(.*)\\' is not defined-g-args1]
joblib/test/test_func_inspect.py::test_filter_args_error_msg[ValueError-Wrong number of arguments-h-args2]
joblib/test/test_memory.py::test_memory_func_with_kwonly_args
joblib/test/test_memory.py::test_memory_func_with_kwonly_args
  /builddir/build/BUILDROOT/python-joblib-0.13.2-5.20190824gitfd585ce.fc32.noarch/usr/lib/python3.8/site-packages/joblib/func_inspect.py:192: DeprecationWarning: `formatargspec` is deprecated since Python 3.5. Use `signature` and the `Signature` object directly
    arg_spec_str = inspect.formatargspec(*arg_spec_for_format)
joblib/test/test_hashing.py::test_bound_cached_methods_hash
joblib/test/test_hashing.py::test_bound_cached_methods_hash
  /builddir/build/BUILDROOT/python-joblib-0.13.2-5.20190824gitfd585ce.fc32.noarch/usr/lib/python3.8/site-packages/joblib/test/test_hashing.py:69: DeprecationWarning: The 'cachedir' parameter has been deprecated in version 0.12 and will be removed in version 0.14.
  You provided "cachedir='/tmp/pytest-of-mockbuild/pytest-0/test_bound_cached_methods_hash0'", use "location='/tmp/pytest-of-mockbuild/pytest-0/test_bound_cached_methods_hash0'" instead.
    mem = Memory(cachedir=cachedir)
joblib/test/test_hashing.py::test_dict_hash
  /builddir/build/BUILDROOT/python-joblib-0.13.2-5.20190824gitfd585ce.fc32.noarch/usr/lib/python3.8/site-packages/joblib/test/test_hashing.py:69: DeprecationWarning: The 'cachedir' parameter has been deprecated in version 0.12 and will be removed in version 0.14.
  You provided "cachedir='/tmp/pytest-of-mockbuild/pytest-0/test_dict_hash0'", use "location='/tmp/pytest-of-mockbuild/pytest-0/test_dict_hash0'" instead.
    mem = Memory(cachedir=cachedir)
joblib/test/test_hashing.py::test_set_hash
  /builddir/build/BUILDROOT/python-joblib-0.13.2-5.20190824gitfd585ce.fc32.noarch/usr/lib/python3.8/site-packages/joblib/test/test_hashing.py:69: DeprecationWarning: The 'cachedir' parameter has been deprecated in version 0.12 and will be removed in version 0.14.
  You provided "cachedir='/tmp/pytest-of-mockbuild/pytest-0/test_set_hash0'", use "location='/tmp/pytest-of-mockbuild/pytest-0/test_set_hash0'" instead.
    mem = Memory(cachedir=cachedir)
joblib/test/test_memory.py::test_memory_integration
  /builddir/build/BUILDROOT/python-joblib-0.13.2-5.20190824gitfd585ce.fc32.noarch/usr/lib/python3.8/site-packages/joblib/test/test_memory.py:103: UserWarning: Compressed results cannot be memmapped
    memory = Memory(location=tmpdir.strpath, verbose=10,
joblib/test/test_memory.py::test_memory_integration
  /builddir/build/BUILDROOT/python-joblib-0.13.2-5.20190824gitfd585ce.fc32.noarch/usr/lib/python3.8/site-packages/joblib/memory.py:131: UserWarning: Compressed items cannot be memmapped in a filesystem store. Option will be ignored.
    obj.configure(location, verbose=verbose,
joblib/test/test_memory.py::test_memory_integration
  /usr/lib/python3.8/contextlib.py:113: UserWarning: mmap_mode "r" is not compatible with compressed file /tmp/pytest-of-mockbuild/pytest-0/test_memory_integration0/joblib/joblib/test/test_memory/f/6946afe22f5c7e3ab08058d3ff5993cf/output.pkl. "r" flag will be ignored.
    return next(self.gen)
joblib/test/test_numpy_pickle.py::test_numpy_persistence[False]
joblib/test/test_numpy_pickle.py::test_numpy_persistence[True]
joblib/test/test_numpy_pickle.py::test_numpy_persistence[0]
joblib/test/test_numpy_pickle.py::test_numpy_persistence[3]
joblib/test/test_numpy_pickle.py::test_numpy_persistence[zlib]
  /builddir/build/BUILDROOT/python-joblib-0.13.2-5.20190824gitfd585ce.fc32.noarch/usr/lib/python3.8/site-packages/joblib/test/test_numpy_pickle.py:169: PendingDeprecationWarning: the matrix subclass is not the recommended way to represent matrices or deal with linear algebra (see https://docs.scipy.org/doc/numpy/user/numpy-for-matlab-users.html). Please adjust your code to use regular ndarray.
    for obj in (np.matrix(np.zeros(10)),
joblib/test/test_numpy_pickle.py::test_compressed_pickle_dump_and_load
  /builddir/build/BUILDROOT/python-joblib-0.13.2-5.20190824gitfd585ce.fc32.noarch/usr/lib/python3.8/site-packages/joblib/test/test_numpy_pickle.py:369: PendingDeprecationWarning: the matrix subclass is not the recommended way to represent matrices or deal with linear algebra (see https://docs.scipy.org/doc/numpy/user/numpy-for-matlab-users.html). Please adjust your code to use regular ndarray.
    np.matrix([0, 1, 2], dtype=np.dtype('<i8')),
joblib/test/test_numpy_pickle.py::test_compressed_pickle_dump_and_load
  /builddir/build/BUILDROOT/python-joblib-0.13.2-5.20190824gitfd585ce.fc32.noarch/usr/lib/python3.8/site-packages/joblib/test/test_numpy_pickle.py:370: PendingDeprecationWarning: the matrix subclass is not the recommended way to represent matrices or deal with linear algebra (see https://docs.scipy.org/doc/numpy/user/numpy-for-matlab-users.html). Please adjust your code to use regular ndarray.
    np.matrix([0, 1, 2], dtype=np.dtype('>i8')),
joblib/test/test_numpy_pickle.py::test_file_handle_persistence
  /builddir/build/BUILDROOT/python-joblib-0.13.2-5.20190824gitfd585ce.fc32.noarch/usr/lib/python3.8/site-packages/joblib/test/test_numpy_pickle.py:652: PendingDeprecationWarning: the matrix subclass is not the recommended way to represent matrices or deal with linear algebra (see https://docs.scipy.org/doc/numpy/user/numpy-for-matlab-users.html). Please adjust your code to use regular ndarray.
    np.matrix([0, 1, 2])]
joblib/test/test_numpy_pickle.py::test_in_memory_persistence
  /builddir/build/BUILDROOT/python-joblib-0.13.2-5.20190824gitfd585ce.fc32.noarch/usr/lib/python3.8/site-packages/joblib/test/test_numpy_pickle.py:686: PendingDeprecationWarning: the matrix subclass is not the recommended way to represent matrices or deal with linear algebra (see https://docs.scipy.org/doc/numpy/user/numpy-for-matlab-users.html). Please adjust your code to use regular ndarray.
    np.matrix([0, 1, 2])]
joblib/test/test_parallel.py::test_nested_loop[threading-multiprocessing]
  /builddir/build/BUILDROOT/python-joblib-0.13.2-5.20190824gitfd585ce.fc32.noarch/usr/lib/python3.8/site-packages/joblib/parallel.py:672: UserWarning: Multiprocessing-backed parallel loops cannot be nested below threads, setting n_jobs=1
    n_jobs = self._backend.configure(n_jobs=self.n_jobs, parallel=self,
joblib/test/test_parallel.py::test_nested_loop[threading-loky]
joblib/test/test_parallel.py::test_nested_loop[threading-back_compat_backend]
  /builddir/build/BUILDROOT/python-joblib-0.13.2-5.20190824gitfd585ce.fc32.noarch/usr/lib/python3.8/site-packages/joblib/parallel.py:672: UserWarning: Loky-backed parallel loops cannot be nested below threads, setting n_jobs=1
    n_jobs = self._backend.configure(n_jobs=self.n_jobs, parallel=self,
joblib/test/test_parallel.py::test_retrieval_context
joblib/test/test_parallel.py::test_retrieval_context
joblib/test/test_parallel.py::test_retrieval_context
joblib/test/test_parallel.py::test_retrieval_context
joblib/test/test_parallel.py::test_retrieval_context
  /builddir/build/BUILDROOT/python-joblib-0.13.2-5.20190824gitfd585ce.fc32.noarch/usr/lib/python3.8/site-packages/joblib/parallel.py:267: DeprecationWarning: check_pickle is deprecated in joblib 0.12 and will be removed in 0.13
    warnings.warn('check_pickle is deprecated in joblib 0.12 and will be'
joblib/test/test_testing.py::test_check_subprocess_call_timeout
  /builddir/build/BUILDROOT/python-joblib-0.13.2-5.20190824gitfd585ce.fc32.noarch/usr/lib/python3.8/site-packages/joblib/testing.py:50: UserWarning: Timeout running ['/usr/bin/python3', '-c', 'import time\nimport sys\nprint("before sleep on stdout")\nsys.stdout.flush()\nsys.stderr.write("before sleep on stderr")\nsys.stderr.flush()\ntime.sleep(1.1)\nprint("process should have be killed before")\nsys.stdout.flush()']
    warnings.warn("Timeout running {}".format(cmd))
-- Docs: https://docs.pytest.org/en/latest/warnings.html
= 8 failed, 1122 passed, 18 skipped, 1 deselected, 32 warnings in 222.49 seconds =
```

And the latest failure comes from:

bpo-36867: The multiprocessing.resource_tracker replaces the multiprocessing.semaphore_tracker module. Other than semaphores, resource_tracker also tracks shared_memory segments.
https://bugs.python.org/issue36867

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -4,7 +4,8 @@ Latest changes
 Release 0.14.0
 --------------
 
-- Add official support for Python 3.8 (updated tests).
+- Add official support for Python 3.8: fixed protocol number in `Hasher`
+  and updated tests.
 
 - Warn users that they should never use `joblib.load` with files from
   untrusted sources. Fix security related API change introduced in numpy

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -4,6 +4,8 @@ Latest changes
 Release 0.14.0
 --------------
 
+- Add official support for Python 3.8 (updated tests).
+
 - Warn users that they should never use `joblib.load` with files from
   untrusted sources. Fix security related API change introduced in numpy
   1.6.3 that would prevent using joblib with recent numpy versions.

--- a/joblib/hashing.py
+++ b/joblib/hashing.py
@@ -58,8 +58,7 @@ class Hasher(Pickler):
         self.stream = io.BytesIO()
         # By default we want a pickle protocol that only changes with
         # the major python version and not the minor one
-        protocol = (pickle.DEFAULT_PROTOCOL if PY3_OR_LATER
-                    else pickle.HIGHEST_PROTOCOL)
+        protocol = 3 if PY3_OR_LATER else 2
         Pickler.__init__(self, self.stream, protocol=protocol)
         # Initialise the hash obj
         self._hash = hashlib.new(hash_name)

--- a/joblib/test/test_module.py
+++ b/joblib/test/test_module.py
@@ -26,7 +26,7 @@ def test_no_start_method_side_effect_on_import():
 
 @pytest.mark.skipif(PY27, reason="Need python3.3+")
 def test_no_semaphore_tracker_on_import():
-    # check that importing joblib does not implicitly spawn a ressource tracker
+    # check that importing joblib does not implicitly spawn a resource tracker
     # or a semaphore tracker
     code = """if True:
         import joblib
@@ -35,6 +35,9 @@ def test_no_semaphore_tracker_on_import():
         # start_method is already set.
         msg = "multiprocessing.semaphore_tracker has been spawned on import"
         assert semaphore_tracker._semaphore_tracker._fd is None, msg"""
+    if sys.version_info >= (3, 8):
+        # semaphore_tracker was renamed in Python 3.8:
+        code = code.replace("semaphore_tracker", "resource_tracker")
     check_subprocess_call([sys.executable, '-c', code])
 
 

